Assets/MRTK/Tests/PlayModeTests/ManipulationHandlerTests.cs

```diff
@@ -1256,6 +1256,128 @@ public IEnumerator ManipulationHandlerEventsBothHands()
+            yield return null;
+
+            // Two Handed
+            manipHandler.ManipulationType = ManipulationHandler.HandMovementType.TwoHandedOnly;
+
+            // Only testing move/rotate centroid position
+            manipHandler.TwoHandedManipulationType = ManipulationHandler.TwoHandedManipulation.MoveRotate;
+
+            // Only testing near manipulation
+            manipHandler.AllowFarManipulation = false;
+
+            int manipulationStartedCount = 0;
+            int manipulationEndedCount = 0;
+            manipHandler.OnManipulationStarted.AddListener((med) => manipulationStartedCount++);
+            manipHandler.OnManipulationEnded.AddListener((med) => manipulationEndedCount++);
+
+            // Grab the box.
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return leftHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+
+            // Should not have moved (yet!)
+            TestUtilities.AssertAboutEqual(testObject.transform.position, initialObjectPosition, $"Object moved when it shouldn't have! Position: {testObject.transform.position:F5}", 0.00001f);
+
+            // The ObjectManipulator should recognize that we've begun manipulation.
+            Assert.IsTrue(manipulationStartedCount == 1);
+
+            // Move both hands outwards; the object may be scaled but the position should remain the same.
+            yield return rightHand.MoveTo(new Vector3(0.2f, -0.1f, 0.8f));
+            yield return leftHand.MoveTo(new Vector3(-0.2f, -0.1f, 0.8f));
+
+            // Should *still* not have moved!
+            TestUtilities.AssertAboutEqual(testObject.transform.position, initialObjectPosition, $"Object moved when it shouldn't have! Position: {testObject.transform.position:F5}", 0.00001f);
+
+            // Manipulation should not yet have ended.
+            Assert.IsTrue(manipulationEndedCount == 0);
+
+            // Get the grab points before we rotate the hands.
+            var leftGrabPoint = manipHandler.GetPointerGrabPoint(leftHand.GetPointer<SpherePointer>().PointerId);
+            var rightGrabPoint = manipHandler.GetPointerGrabPoint(rightHand.GetPointer<SpherePointer>().PointerId);
+            var originalCentroid = (leftGrabPoint + rightGrabPoint) / 2.0f;
+
+            // List of test conditions for test fuzzing.
+            // (position, rotation)
+            List<(Vector3, Vector3)> testConditions = new List<(Vector3, Vector3)>
+            {
+                (new Vector3(0, 90, 0), new Vector3(0.2f, -0.1f, 0.8f)),
+                (new Vector3(25, 30, 45), new Vector3(0.3f, -0.2f, 0.7f)),
+                (new Vector3(75, 140, 0), new Vector3(0.1f, -0.1f, 0.8f)),
+                (new Vector3(10, 90, 20), new Vector3(0.5f, -0.2f, 0.5f)),
+                (new Vector3(45, 110, 0), new Vector3(0.3f, -0.1f, 0.8f))
+            };
+
+            // Fuzz test.
+            foreach (var testCondition in testConditions)
```

Comment:
Sanity checking this -- how many seconds does this add to the playmode tests? Looks like it should be fine, but want to make sure we are aiming to add < 1 second of test time whenever we add new test. Just a general guidelines, in some cases we may need more time.

Reply:
It is quite fast, < 1s I believe. I can reduce the # of frames that are used for the MoveHand calls.